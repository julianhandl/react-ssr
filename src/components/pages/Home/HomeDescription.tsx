import React from 'react';

import './HomeDescriptionStyles.scss';

const HomeDescription : React.FunctionComponent = () => {
    return (
        <article className="home-description__wrapper">
            <div className="container">
                <div className="home-description">
                    <h1>Wir machen es besser!</h1>
                    <p>Mieten war immer mühselig. Unübersichtliche Equipment Listen. Preise ohne Steuer. Meist nur als Firma zugänglich. Auch als Privatperson ist Mieten sinnvoll. Die Top Gründe warum Sie etwas mieten sollten statt zu kaufen:</p>
                    <ul>
                        <li>Es kostet Ihnen nur einen Bruchteil des Kaufpreises.</li>
                        <li>Sie benötigen keinen Stauraum für die Lagerung.</li>
                        <li>Sie müssen nichts reparieren oder warten.</li>
                    </ul>
                    <p>Wir machen Ihnen das Mieten so einfach wie möglich. Suchen Sie ihr Equipment online aus oder lassen Sie sich von uns beraten.</p>
                </div>
            </div>
        </article>
    );
};

export default HomeDescription;