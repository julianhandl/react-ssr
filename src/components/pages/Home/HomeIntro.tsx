import React from 'react';
import {packageMusik, partyZelt} from "../../../../../data/products";

import './HomeIntroStyles.scss';
import { Link } from 'react-router-dom';
import { urls } from '../../../Routes';

const HomeIntro : React.FunctionComponent = () => {
    const musik = packageMusik.variants[Object.keys(packageMusik.variants)[0]];
    const zelt = partyZelt.variants[Object.keys(partyZelt.variants)[0]];

    return (
        <div className="container">
            <article className="home-intro">
                <section className="home-intro__slogan">
                    <h1>
                        <span>Mieten Sie</span>
                        <small>was sonst verstauben würde</small>
                    </h1>
                    <ul>
                        <li>Ihr persönlicher Partyservice</li>
                        <li>Einfache online Übersicht</li>
                        <li>Fixe Preise und online Buchung</li>
                        {/* <li>Alles für Privatpersonen</li> */}
                    </ul>
                    {/*
                    <div className="home-intro__action">
                        <Link to={urls.musik} className="home-intro__action-button home-intro__action-button--musik">Musik mieten</Link>
                        <Link to={urls.zelt} className="home-intro__action-button home-intro__action-button--zelt">Zelt mieten</Link>
                    </div>
                    */}
                </section>
                <div className="home-intro__boxes">
                    <Link to={urls.musik} className="home-intro__box">
                        <h2>{packageMusik.shortTitle}</h2>
                        <p>{packageMusik.description}</p>
                        <div className="home-intro__box-price">
                            <small>ab</small>
                            <span>{(musik.priceCents / 100).toFixed(0)}€</span>
                            <small className="home-intro__box-price-day">/ Tag</small>
                        </div>
                    </Link>
                    <Link to={urls.zelt} className="home-intro__box">
                        <h2>{partyZelt.shortTitle}</h2>
                        <p>{partyZelt.description}</p>
                        <div className="home-intro__box-price">
                            <small>ab</small>
                            <span>{(zelt.priceCents / 100).toFixed(0)}€</span>
                            <small className="home-intro__box-price-day">/ Tag</small>
                        </div>
                    </Link>
                </div>
            </article>
        </div>
    );
};

export default HomeIntro;