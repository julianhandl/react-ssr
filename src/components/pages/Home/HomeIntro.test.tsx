import React from "react";
import { mount } from "enzyme";
import HomeDescription from "./HomeDescription";
import HomeIntro from "./HomeIntro";
import { packageMusik, partyZelt } from "../../../../../data/products";

describe("HomeIntro", () => {
    const wrapper = mount(<div><HomeIntro /></div>)

    it("should render correctly", () => {
        expect(wrapper.find(".container").length).toBe(1);
        expect(wrapper.find(".container > .home-intro").length).toBe(1);

        // slogan
        expect(wrapper.find(".container > .home-intro > .home-intro__slogan").length).toBe(1);
        const slogan = wrapper.find(".home-intro__slogan");
        expect(slogan).toMatchSnapshot();

        //boxes
        expect(wrapper.find(".container > .home-intro > .home-intro__boxes").length).toBe(1);
        const boxes = wrapper.find(".home-intro__box");
        expect(boxes.length).toBe(2);

        const musik = packageMusik.variants[Object.keys(packageMusik.variants)[0]];
        const zelt = partyZelt.variants[Object.keys(partyZelt.variants)[0]];

        // musik
        expect(boxes.at(0).find("h2").length).toBe(1);
        expect(boxes.at(0).find("h2").text()).toBe(packageMusik.shortTitle);
        expect(boxes.at(0).find("p").length).toBe(1);
        expect(boxes.at(0).find("p").text()).toBe(packageMusik.description);
        expect(boxes.at(0).find(".home-intro__box-price").length).toBe(1);
        expect(boxes.at(0).find(".home-intro__box-price span").text()).toBe(`${(musik.priceCents / 100).toFixed(0)}â‚¬`);

    })
});