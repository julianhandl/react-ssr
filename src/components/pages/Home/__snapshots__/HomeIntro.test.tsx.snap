// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeIntro should render correctly 1`] = `
<section
  className="home-intro__slogan"
>
  <h1>
    <span>
      Mieten Sie
    </span>
    <small>
      was sonst verstauben würde
    </small>
  </h1>
  <ul>
    <li>
      Einfache online Übersicht
    </li>
    <li>
      Fixe Preise und online Buchung
    </li>
    <li>
      Alles für Privatpersonen
    </li>
  </ul>
</section>
`;
